### ISSUE TITLE
'Models must be owned by only a single document' issue

### ISSUE THREAD

**Issue by axil** [bokeh/jupyter_bokeh#167]
The following code that  intermixes two BokehModel instances with an ipywidget instance 
```python
import bokeh.plotting as bp
import bokeh.models as bm
import ipywidgets as ipw
from jupyter_bokeh import BokehModel
p = bp.figure()
src = bm.ColumnDataSource(data={'x': [1,2], 'y': [3,4]}) 
p.line(source=src)
radio = bm.RadioGroup(labels=['a', 'b'], active=0)
label = ipw.Label('c')
callback = bm.CustomJS(args=dict(data_src=src))
radio.js_on_change('active', callback)
ipw.VBox([
    BokehModel(p),
    BokehModel(radio), 
    label,
])
```

generates the following error:

    RuntimeError: Models must be owned by only a single document, Selection(id='1732', ...) is already in a doc

Supposedly the problem is with `jupyter_bokeh` because

```python
import bokeh.layouts as bl
bp.show(bl.column(p, radio))
```

works fine.

**Comment by mattpap** [comment]
There is a potential workaround for the "single document" error:
```py
doc = Document()
doc.add_root(p)
doc.add_root(radio)
ipw.VBox([
    BokehModel(p),
    ipw.HBox([
          BokehModel(radio), 
          label,
    ]),
])
```
This way `p` and `radio` will share a document and `BokehModel` will not attempt to attach a new one under such setup. However, that only solves part of the problem. The other issue is that each `BokehModel(...)` is an independent embedding point, which is currently undefined behavior when using the same document. Consider each `BokehModel(...)` as if calling `show()` multiple times with the same document. However, I have some ideas how this could be alleviated.

**Comment by axil** [comment]
Yes, it helps with the ownership issue. Yet this code displays both widgets twice and generates the following error

![image](https://user-images.githubusercontent.com/170910/186309158-46a15302-9665-49c7-96cf-e7a28fbf7928.png)

for every model (a total of 8 times). Do you know how to deal with it?
